5150: Introduce Ding SEO module

holt83 opened the pull request:
#### Link to issue

https://platform.dandigbib.org/issues/5150

#### Description

Introduce Ding SEO module which provides:

- JSON-LD structured data on content
- JSON-LD structured data on ting_object materials (books and movies).
- Easy search console setup
- XMLSitemap module integration

#### Screenshot of the result

If your change affects the user interface you should include a screenshot of the result with the pull request.

#### Checklist

- [ ] My complies with [our coding guidelines](../docs/code_guidelines.md).
- [ ] My code passes our static analysis suite. If not then I have added a comment explaining why this change should be exempt from the code standards and process.
- [ ] My code passes our continuous integration process. If not then I have added a comment explaining why this change should be exempt from the code standards and process.

#### Additional comments or questions

If you have any further comments or questions for the reviewer them please add them here.

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: f43ac6caa3443454b273c7e3734d2c4e51eb421e
  * The subject should be kept around 50 characters long. The subject must not be more than 69 characters long. Strive for about 50 characters.

 * Commit: 5446a6aef90c0f33f22dd159e0e7f903ac58e733
  * The subject should be kept around 50 characters long. The subject must not be more than 69 characters long. Strive for about 50 characters.

 * Commit: 0df62eda90b80baa1f455485582505be266af52d
  * The subject should be kept around 50 characters long. The subject must not be more than 69 characters long. Strive for about 50 characters.

 * Commit: c6a062f67cefd2781e50732370cab7166468b712
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: f43ac6caa3443454b273c7e3734d2c4e51eb421e
  * The subject should be kept around 50 characters long. The subject must not be more than 69 characters long. Strive for about 50 characters.

 * Commit: 5446a6aef90c0f33f22dd159e0e7f903ac58e733
  * The subject should be kept around 50 characters long. The subject must not be more than 69 characters long. Strive for about 50 characters.

 * Commit: 0df62eda90b80baa1f455485582505be266af52d
  * The subject should be kept around 50 characters long. The subject must not be more than 69 characters long. Strive for about 50 characters.

 * Commit: c6a062f67cefd2781e50732370cab7166468b712
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: f43ac6caa3443454b273c7e3734d2c4e51eb421e
  * The subject should be kept around 50 characters long. The subject must not be more than 69 characters long. Strive for about 50 characters.

 * Commit: 5446a6aef90c0f33f22dd159e0e7f903ac58e733
  * The subject should be kept around 50 characters long. The subject must not be more than 69 characters long. Strive for about 50 characters.

 * Commit: 0df62eda90b80baa1f455485582505be266af52d
  * The subject should be kept around 50 characters long. The subject must not be more than 69 characters long. Strive for about 50 characters.

 * Commit: c6a062f67cefd2781e50732370cab7166468b712
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: f43ac6caa3443454b273c7e3734d2c4e51eb421e
  * The subject should be kept around 50 characters long. The subject must not be more than 69 characters long. Strive for about 50 characters.

 * Commit: 5446a6aef90c0f33f22dd159e0e7f903ac58e733
  * The subject should be kept around 50 characters long. The subject must not be more than 69 characters long. Strive for about 50 characters.

 * Commit: 0df62eda90b80baa1f455485582505be266af52d
  * The subject should be kept around 50 characters long. The subject must not be more than 69 characters long. Strive for about 50 characters.

 * Commit: c6a062f67cefd2781e50732370cab7166468b712
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

kasperg reviewed (changes requested):
:+1: Lots of good stuff here but also a lot of details to nerd out on.

Some general topics that I would like to see addressed in some way:

- Actual values set in the metadata for content.
- Some background info about the decision to load JSON-LD data for objects asynchronously. If this implementation is to remain then it at leasts needs some commenting about why it is needed.
- The data provided for objects in regard to types as well as providers.
- Whether the handling of tokens can be update to reduce the amount of custom code needed

In general I had also expected the PR to contain some code to support objects and coillections in the XML sitemap module. I could not spot any. Am I missing something?

Also note the build failure caused by PHP 5.6 compatibility problems. Also the PR does not merge cleanly in the current state.

agnetemoos commented on modules/ding_event/ding_event.module:
```diff
@@ -195,7 +195,8 @@ function ding_event_views_query_alter(&$view, &$query) {
   // set up in views, but we need this work with the exposed filters on the
   // ding_event_list also.
-  if ($view->name == 'ding_event' && in_array($view->current_display, array('ding_event_list', 'ding_event_list_frontpage'))) {
+  if ($view->name == 'ding_event' && in_array($view->current_display, array('ding_event_list', 'ding_event_list_frontpage'))
+    && in_array(current_path(), array('ding_frontpage', 'arrangementer'))) {
```
The changes to ding_event.features.field_instance.inc, ding_event.module, ding_event.pages_default.inc and ding_news.pages_default.inc were introduced to accomplish this goal: The news- and event category pages are excellent hosts of well-written SEO texts that are reusable for more than one season. Imagine at well-written text about the blessings of "børneteater". Could be an effective Google landingpage for years without modification. So if the category-pages could be found by Google (ding_event.features.field_instance.inc)  and a text could be added at the bottom (ding_event.pages_default.inc and ding_news.pages_default.inc) we would be home safe.
BUT in the meantime sections have been introduced. With this tool we can easily accomplish the same. So I suggest we drop all changes to these four files.

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

holt83 commented on modules/ding_seo/ding_seo.features.metatag.inc:
```diff
@@ -0,0 +1,145 @@
+    'instance' => 'node:ding_news',
+    'disabled' => FALSE,
+    'config' => array(
+      'schema_article.@type' => array(
+        'value' => 'NewsArticle',
```
I have looked into it and it seems all Google's examples for the Article type uses NewsArticle:

https://developers.google.com/search/docs/advanced/structured-data/article

I think that's why NewsArticle was chosen.

We could easily change the type to Article, but maybe it's best to follow Google's examples?

holt83 commented on modules/ding_seo/ding_seo.tokens.inc:
```diff
@@ -0,0 +1,286 @@
+      }
+    }
+
+    // Get logo dimensions.
+    global $base_url;
+    drupal_theme_initialize();
+    $logo_url = theme_get_setting('logo');
+    if (!empty($logo_url)) {
+      // Get relative internal for getimagesize. Add 1 to remove leading slash.
+      $logo_path = drupal_substr($logo_url, drupal_strlen($base_url) + 1);
+      $logo_info = getimagesize($logo_path);
+      $logo_width = $logo_info[0];
+      $logo_height = $logo_info[1];
+    }
```
We also need the dimensions though, but it would be nice to avoid this code.

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

holt83 commented on modules/ding_seo/ding_seo.features.metatag.inc:
```diff
@@ -0,0 +1,145 @@
+        'value' => '[node:field_ding_library_phone_number]',
+      ),
+      'schema_organization.url' => array(
+        'value' => '[site:url]',
```
This has been changed to `[node:url]` for both `@id` and `url`

holt83 commented on modules/ding_base/ding_base.info:
```diff
@@ -108,6 +108,37 @@ features[variable][] = jquery_update_compression_type
 features[variable][] = jquery_update_jquery_cdn
 features[variable][] = jquery_update_jquery_migrate_enable
 features[variable][] = jquery_update_jquery_version
+features[variable][] = metatag_entity_type_groups_file_audio
```
Entity type group definitions has been moved to `ding_seo` module.

holt83 commented on modules/ding_seo/ding_seo.module:
```diff
@@ -0,0 +1,228 @@
+  if (!isset($schema_wrapper) || !$schema_type = $schema_wrapper->getSchemaType()) {
+    print '';
+    drupal_exit();
+  }
```
Now returning 404 if no wrapper was found.

holt83 commented on modules/ding_seo/ding_seo.module:
```diff
@@ -0,0 +1,228 @@
+    'file' => 'ding_seo.admin.inc',
+  ];
+
+  $items['ding_seo/jsonld/ting_object/%'] = [
+    'title' => 'Ding SEO JSON-LD',
+    'page callback' => 'ding_seo_jsonld_ting_object_callback',
+    'page arguments' => [3],
+    'access arguments' => ['access content'],
+  ];
```
According to Google they support loading structured data async with javascript:

https://developers.google.com/search/docs/advanced/structured-data/generate-structured-data-with-javascript

holt83 replied:
The reasons for switching to custom async JSON-LD was to be able to check for covers and availability/reservability asynchronously. 

Was also having some issues with schema_meatag module and JSON-LD ting obejcts. Especially with work examples on books, which needed some quite complicated hacks to get working correctly.

holt83 commented on modules/ding_seo/ding_seo.install:
```diff
@@ -0,0 +1,14 @@
+ */
+function ding_seo_install() {
+  // Ensure the ting_object global metatag config is enabled.
+  variable_set('metatag_enable_ting_object', TRUE);
```
Removed

holt83 commented on modules/ding_seo/ding_seo.tokens.inc:
```diff
@@ -0,0 +1,286 @@
+  // Define tokens supported on ding_seo_node_tokens.
+  $info['tokens']['ding_seo_node_tokens']['location_name'] = array(
+    'name' => t("Location Name"),
+    'description' => t('The location is the '),
```
The description has been updated

holt83 commented on modules/ding_seo/ding_seo.tokens.inc:
```diff
@@ -0,0 +1,286 @@
+  $info['tokens']['ding_seo_node_tokens']['event_enddate'] = [
+    'name' => t("Event Enddate"),
+    'description' => t('Event enddate in YYYY-MM-DD HH:MM::SS format'),
+  ];
```
I have removed several of our custom node tokens and replaced with field tokens defined by the entity module.

holt83 commented on modules/ding_seo/src/TingObjectSchemaWrapperBase.php:
```diff
@@ -0,0 +1,122 @@
+    // First check if this is a known negative.
+    if (cache_get('ting_covers:' . $ting_object_id)) {
+      return $this->image_url;
+    }
```
Now using `ting_covers_get()`

holt83 commented on modules/opensearch/src/OpenSearchTingObjectSchemaWrapper.php:
```diff
@@ -0,0 +1,229 @@
+   */
+  public function getDescription() {
+    return $this->ting_object->getAbstract();
+  }
```
Moved several function to base class.

holt83 commented on modules/ding_seo/ding_seo.module:
```diff
@@ -0,0 +1,228 @@
+    case 'Movie':
+      $node += [
+        '@id' => $schema_wrapper->getObjectURL(),
+        'url' => $schema_wrapper->getObjectURL(),
```
I have moved more of the properties to common.

We still need some special handling for books since they have nested work examples with some additional props and some them should be repeated if they differ from top-level work.

Also different kind of contributors for movies and books.

holt83 commented on modules/ding_seo/ding_seo.module:
```diff
@@ -0,0 +1,228 @@
+    case 'Book':
+      $authors = $schema_wrapper->getAuthors();
+      $node += [
+        '@id' => $schema_wrapper->getCollectionURL(),
+        'url' => $schema_wrapper->getCollectionURL(),
```
have looked into it and according to Google these properties **can** be the same as the ones on work examples, which i believe was my initial concern:

https://developers.google.com/search/docs/advanced/structured-data/book#book-work

So both if these properties now use object URL for books, work examples and movies.

holt83 commented on modules/ding_seo/ding_seo.module:
```diff
@@ -0,0 +1,228 @@
+      'url' => $image_url,
+    ];
+
+    // TODO: Find out if it's ok to omit dimensions or we should just ditch
+    // it all together if not found. We know there's a cover, but it might
+    // not be fecthed yet so we can check dimensions.
+    if ($image_dimensions = $schema_wrapper->getImageDimensions()) {
+      $node['image'] += [
+        'width' => "$image_dimensions[0]",
+        'height' => "$image_dimensions[1]",
+      ];
+    }
```
Removed.. they are optional.

holt83 commented on modules/ding_seo/ding_seo.js:
```diff
@@ -0,0 +1,18 @@
+ * Ding SEO asynchronous JSON-LD.
+ */
+
+(function($) {
+  "use strict";
+
+  // TODO: Add comment explaining why we're not using Drupal behaviors.
+  $(function() {
+    var tingObjectId = Drupal.settings.dingSeo.tingObjectId;
+    $.get(`/ding_seo/jsonld/ting_object/${tingObjectId}`, function(jsonld) {
```
Updated comment:

```
  // Bypass Drupal behaviors since we just need to insert a script in header
  // once and have no need to interact with other elements on the page.
```

At least i believe that was the reason :)

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

holt83 commented:
Updates in PR:

- Code review fixes
- Added BorrowAction to "Book" and "Movie" schema.org types
- Added sameAs bibdk link to "Book" and "Movie"
- Added opengraph tags to ding_news, ding_page and ding_event
- Updated metatag module (so we can use our community patch without issues).
- Added ting_object integration to sitemap module.
- Scrutinizer fixes.

holt83 commented:
The last Scrutinizer error can be ignored:

"Only string literals should be passed to t() where possible"

It should be fine, since the strings we pass to should be translated. And if not it's not a big deal. We just don't show `inLanguage` for that material. It should be very rare though since most language should be translatable and most of them will be "Danish".

holt83 commented:
Appearently we need to embed most of the library JSON-LD in borrow actions, for Google to accept it in https://search.google.com/test/rich-results.

This presents a challenge since this is normally generated by schema_metatag module on page load.

I'm working on a  small "fix", and will probably update the PR tomorrow. 

The rest of PR can still be reviewed.

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

kasperg reviewed (changes requested):
:+1: Lots of code, lots of comments :smile:.

The changes regarding JSON-LD look good. I have a few comments and corrections here and there. I hope they are manageable to address.

The situation regarding the sitemap is more tricky. Grocking the current implementation took me quite a while. It started with comments addressing minor bits here and there and led to some greater frustrations.

Take a look at it and we can discuss what is possible to change at this point.

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

holt83 commented on modules/ding_seo/ding_seo.features.metatag.inc:
```diff
@@ -0,0 +1,245 @@
+    'disabled' => FALSE,
+    'config' => array(
+      'title' => array(
+        'value' => '[node:title] | [current-page:pager][site:name]',
```
This is a special token defined by the metatag module which insert a pager in the title if present.

I needs to be without whitespace to work correctly.

The setting for this feature can be found at: /admin/config/search/metatags/settings

holt83 commented on modules/ding_seo/ding_seo.features.metatag.inc:
```diff
@@ -0,0 +1,245 @@
+      ),
+      'og:country_name' => array(
+        'value' => '[node:ding_seo_node_tokens:address_country]',
+      ),
```
You're right. Removed

holt83 commented on modules/ding_seo/ding_seo.features.metatag.inc:
```diff
@@ -0,0 +1,245 @@
+      ),
+      'schema_event.@id' => array(
+        'value' => '[node:url]',
+      ),
```
Good catch. The URL will change everytime title is changed, so it's actually not appropiate to use. Changed to using node ID.

holt83 commented on modules/ding_seo/ding_seo.features.metatag.inc:
```diff
@@ -0,0 +1,245 @@
+      ),
+      'schema_event.url' => array(
+        'value' => '[node:url]',
+      ),
```
I don't think it matters. Changes it to use absolute to be consistent and explicit.

holt83 commented on modules/ding_seo/ding_seo.features.metatag.inc:
```diff
@@ -0,0 +1,245 @@
+        'value' => 'a:5:{s:5:"@type";s:12:"Organization";s:3:"@id";s:10:"[site:url]";s:4:"name";s:11:"[site:name]";s:3:"url";s:10:"[site:url]";s:4:"logo";a:4:{s:5:"@type";s:11:"ImageObject";s:3:"url";s:26:"[ding_seo_logo_tokens:url]";s:5:"width";s:28:"[ding_seo_logo_tokens:width]";s:6:"height";s:29:"[ding_seo_logo_tokens:height]";}}',
+      ),
+      'schema_article.publisher' => array(
+        'value' => 'a:5:{s:5:"@type";s:12:"Organization";s:3:"@id";s:10:"[site:url]";s:4:"name";s:11:"[site:name]";s:3:"url";s:10:"[site:url]";s:4:"logo";a:4:{s:5:"@type";s:11:"ImageObject";s:3:"url";s:26:"[ding_seo_logo_tokens:url]";s:5:"width";s:28:"[ding_seo_logo_tokens:width]";s:6:"height";s:29:"[ding_seo_logo_tokens:height]";}}',
```
I actually don't know why it was done this way. Perhaps something with anonymity. It doesn't really make sense though, since the auther is shown on the page anyway. Changed it to use the node author instead.

holt83 commented on modules/ding_seo/ding_seo.features.metatag.inc:
```diff
@@ -0,0 +1,245 @@
+      'schema_article.mainEntityOfPage' => array(
+        'value' => '[current-page:url]',
+      ),
+      'schema_article.about' => array(
+        'value' => '[node:field_ding_news_tags]',
+      ),
```
It looks it's correct to use `[node:field_ding_news_tags]` here. From the description on the metatag settings page:

> Comma separated list of what the article is about, for instance taxonomy terms or categories. Multiple values may be used, separated by a comma. Note: Tokens that return multiple values will be handled automatically.

kasperg replied:
You are right. It makes somewhat sense when you look at it in the schema.org debugger: https://validator.schema.org/#url=https%3A%2F%2Fding-seo-lkadvxi-ton6gvqucinhw.eu-4.platformsh.site%2F%3Fq%3Dnyheder%2Fkort-nyt%2Feventyrere-og-globetrottere

holt83 commented on modules/ding_seo/ding_seo.features.metatag.inc:
```diff
@@ -0,0 +1,245 @@
+    'config' => array(
+      'schema_organization.@type' => array(
+        'value' => 'Library',
+      ),
```
Added opengraph for libraries

holt83 commented on modules/ding_seo/ding_seo.features.metatag.inc:
```diff
@@ -0,0 +1,245 @@
+    'config' => array(
+      'og:type' => array(
+        'value' => 'website',
+      ),
```
Added schema.org JSON-LD for pages. Using the WebPage type, since that is what was available in the schema_metatag module.

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

holt83 commented on modules/ding_seo/ding_seo.features.metatag.inc:
```diff
@@ -0,0 +1,145 @@
+        'value' => '[node:field_ding_library_lead]',
+      ),
+      'schema_organization.image' => array(
+        'value' => 'a:4:{s:5:"@type";s:11:"ImageObject";s:3:"url";s:61:"[node:field_ding_library_title_image:ding_primary_large:path]";s:5:"width";s:62:"[node:field_ding_library_title_image:ding_primary_large:width]";s:6:"height";s:63:"[node:field_ding_library_title_image:ding_primary_large:height]";}',
```
We have used the same image styles as is used for the title image in the content. I think this makes most sense, since editors will use an image that works in that format.

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

holt83 commented on modules/ding_seo/ding_seo.info:
```diff
@@ -0,0 +1,91 @@
+dependencies[] = schema_event
+dependencies[] = schema_organization
+dependencies[] = strongarm
+dependencies[] = xmlsitemap_custom
```
It's enabled so that administrators can add links manually if they are not being picked up automatically for some reason with our current configuration.

kasperg reviewed (changes requested):
:+1: Solid follow up here. I appreciate the changes and the comments where very helpful.

I still believe there are a few outstanding issues.

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

arniloftsson commented on modules/ding_seo/ding_seo.module:
```diff
@@ -351,6 +361,8 @@ function ding_seo_get_library_nodes() {
+
+  $last_run -= 86400;
   // Run once every day.
   if (REQUEST_TIME > $last_run + 86400) {
```
Debug code. Shouldn't be there.

holt83 replied:
Removed

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

GitCop commented:
Please ensure that your commit messages comply with our contribution guidelines.

 * Commit: 15854f7a2ace5e2112aa263756e2898ea5771091
  * Your commit message body contains a line that is longer than 72 characters

Guidelines are available at https://github.com/ding2/ding2/blob/master/CONTRIBUTING.md.

---
This message was auto-generated by https://gitcop.com

kasperg reviewed (approved):
:+1: LGTM
